sunpy/net/dataretriever/sources/gong_synoptic.py

```diff
@@ -0,0 +1,103 @@
+from sunpy.net.dataretriever import GenericClient
+from sunpy.util.scraper import Scraper
+
+__all__ = ['GongSynopticClient']
+
+
+class GongSynopticClient(GenericClient):
+    """
+    Provides access to the Magnetogram products of NSO-GONG synoptic Maps.
+
+    Searches data hosted by the `National Solar Observatory. <gong2.nso.edu/oQR/zqs/>`__
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido, attrs as a
+    >>> results = Fido.search(a.Time("2019/12/31 22:00", "2020/1/1 02:00"),
+    ...                       a.Instrument('GONG'))  #doctest: +REMOTE_DATA
+    >>> results  #doctest: +REMOTE_DATA
+    <sunpy.net.fido_factory.UnifiedResponse object at ...>
+    Results from 1 Provider:
+    <BLANKLINE>
+    3 Results from the GongSynopticClient:
+         Start Time           End Time      Source Instrument Wavelength
+    ------------------- ------------------- ------ ---------- ----------
+    2019-12-31 22:00:00 2020-01-01 02:00:00    NSO       GONG        nan
+    2019-12-31 22:00:00 2020-01-01 02:00:00    NSO       GONG        nan
+    2019-12-31 22:00:00 2020-01-01 02:00:00    NSO       GONG        nan
+    <BLANKLINE>
+    <BLANKLINE>
+
+    """
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Return URL(s) for corresponding timerange.
+
+        Parameters
+        ----------
+        timerange : `~sunpy.time.TimeRange`
+            The time range required.
+
+        Returns
+        -------
+        `list`
+            The URL(s) for the corresponding timerange.
+        """
+        prefix = r'https://gong2.nso.edu/oQR/zqs/%Y%m/mrzqs%y%m%d/'
+        pattern = prefix + r'mrzqs%y%m%dt%H%Mc(\d){4}_(\d){3}\.fits.gz'
+        gongsynoptic_files = Scraper(pattern, regex=True)
+        urls = gongsynoptic_files.filelist(timerange)
+        return urls
+
+    def _makeimap(self):
+        """
+        Helper Function:used to hold information about source.
+        """
+        self.map_['instrument'] = 'GONG'
+        self.map_['source'] = 'NSO'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Answers whether client can service the query.
+
+        Parameters
+        ----------
+        query : list of query objects
+
+        Returns
+        -------
+        bool : Answer as to whether client can service the query.
+        """
+        from sunpy.net import attrs as a
+
+        required = {a.Time, a.Instrument}
+        optional = {a.Physobs, a.gong_synoptic.ExtentType}
+        if not cls.check_attr_types_in_query(query, required, optional):
+            return False
+
+        matches = True
+        for x in query:
+            if isinstance(x, a.Instrument) and x.value.lower() != 'gong':
+                matches = False
+            if isinstance(x, a.Physobs) and x.value.lower() != 'los_magnetic_field':
+                matches = False
+            if isinstance(x, a.gong_synoptic.ExtentType) and x.value.lower() != 'synoptic':
```

Comment:
Why do we have an attr for a thing we only support one type of?

Reply:
I was thinking that we could re-use the VSO `EXTENT_TYPE` keyword here, but it's currently broken: https://github.com/sunpy/sunpy/issues/4048

Reply:
If we don't have any use for this attr I suggest we don't implement it at this time. The extent type for this new client should be set in the `_map` metadata in the client class.

Reply:
I added extentType after this https://github.com/sunpy/sunpy/pull/4055#discussion_r414496406 . We can either remove it too.

Reply:
Is gong data with different extent types available through the VSO? but we can't filter on it because of #4048 ?

Reply:
No, I don't think GONG synoptic maps are on the VSO, but I think it would be good if we can re-purpose the EXTENT_TYPE attr in VSO for search (ie. move it from a vso specific attr to a general one like `Time` or `Instrument`) to avoid creating new attrs.

Reply:
Okay, I did some digging/research and the EXTENT_TYPE attr for VSO is properly broken, so I think including a new attr is fine here.

If it was working then these are the valid values: https://sdac.virtualsolar.org/cgi/show_details?keyword=EXTENT_TYPE - @Cadair do you think it's worth putting the new attr in `sunpy.net.attrs` instead of in `sunpy.net.gong_synoptic.attrs`?

Reply:
Going to resolve this conversation, and move discussion on where this new attr should go to an issue.